pkg/agent/events.go

```diff
@@ -54,30 +55,71 @@ func (ag *Agent) Notify(event testkube.Event) (result testkube.EventResult) {
 	}
 }
 
-func (ag *Agent) RunEventLoop(ctx context.Context) error {
+func (ag *Agent) runEventLoop(ctx context.Context) error {
 	var opts []grpc.CallOption
 	md := metadata.Pairs(apiKey, ag.apiKey)
 	ctx = metadata.NewOutgoingContext(ctx, md)
 
-	//TODO figure out how to retry this method in case of network failure
-	// creates a new Stream from the client side. ctx is used for the lifetime of the stream.
 	stream, err := ag.client.Send(ctx, opts...)
 	if err != nil {
 		ag.logger.Errorf("failed to execute: %w", err)
 		return fmt.Errorf("failed to setup stream: %w", err)
```

Comment:
`github.com/pkg/errors` is used throughout the project, my recommendation is to use that instead of `fmt`